sdk/src/client/api/block_builder/input_selection/requirement/native_tokens.rs

```diff
@@ -59,7 +62,9 @@ impl InputSelection {
         input_native_tokens.merge(minted_native_tokens)?;
         output_native_tokens.merge(melted_native_tokens)?;
 
-        if let Some(burn) = self.burn.as_ref() {
+        if let Some(burn) = self.burn.as_ref().filter(|burn| !burn.native_tokens.is_empty()) {
```

Comment:
Clippy doesn't complain about !is_empty instead of is_some?

Reply:
it's not an option

Reply:
I'm dumb